ui/src/components/crop_image_tool/mod.rs

```diff
@@ -0,0 +1,155 @@
+use dioxus::prelude::*;
+use kit::{
+    elements::{button::Button, range::Range, Appearance},
+    layout::modal::Modal,
+};
+use std::{fs::File, io::Write, path::PathBuf};
+
+#[derive(Props)]
+pub struct Props<'a> {
+    pub large_thumbnail: String,
+    pub on_cancel: EventHandler<'a, ()>,
+    pub on_crop: EventHandler<'a, PathBuf>,
+}
+
+#[allow(non_snake_case)]
+pub fn CropImageModal<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
+    let large_thumbnail = use_ref(cx, || cx.props.large_thumbnail.clone());
+
+    let image_scale: &UseRef<f32> = use_ref(cx, || 1.0);
+    let crop_image = use_state(cx, || true);
+    let adjust_crop_circle_size = include_str!("./adjust_crop_circle_size.js");
+    let cropped_image_pathbuf = use_ref(cx, PathBuf::new);
+    let clicked_button_to_crop = use_state(cx, || false);
+
+    if *clicked_button_to_crop.get() {
+        cx.props.on_crop.call(cropped_image_pathbuf.read().clone());
+        clicked_button_to_crop.set(false);
+        crop_image.set(false);
+    }
+
+    let eval = use_eval(cx);
+    _ = eval(adjust_crop_circle_size);
+
+    return cx.render(rsx!(div {
+        Modal {
+            open: *crop_image.clone(),
+            onclose: move |_| {
+                // Not close if user clicks outside modal
+            },
+            transparent: false, 
+            show_close_button: false,
+            dont_pad: false,
+            div {
+                max_height: "85vh",
+                max_width: "80vw",
+                padding: "16px",
+                onclick: move |_| {},
+                div {
+                    id: "crop-image-topbar", 
+                    background: "var(--secondary)",
+                    height: "70px",
+                    border_radius: "12px",
+                    div {
+                        id: "crop-image-topbar-left",
+                        padding: "16px",
+                        display: "inline-flex",
+                        div {
+                            id: "crop-image-topbar-left-title",
+                            color: "var(--text-color)",
+                            margin_right: "32px",
+                            "Please select the area\n you want to crop"
+                        },
+                        Button {
+                            appearance: Appearance::DangerAlternative,
+                            icon: Shape::XMark,
+                            onpress: move |_| {
+                                cx.props.on_cancel.call(());
+                                crop_image.set(false);
+                            }
+                        },
+                        div {
+                            margin_right: "16px",
+                        }
+                        Button {
+                            appearance: Appearance::Success,
+                            icon: Shape::Check,
+                            onpress: move |_| {
+                                cx.spawn({
```

Comment:
Correct me if im wrong but dont alot of the events in dioxus support async blocks now? If so, maybe we can go without spawning a task but provide the async block for dioxus to poll.

CC @sdwoodbury

Reply:
I think the issue is the `onpress` prop. if the type of the prop was changed, it might be possible to pass an async block to our button element. 

![image](https://github.com/Satellite-im/Uplink/assets/8636602/3c1c2c0a-1397-4071-86d6-8b1f10b577bb)

Reply:
for reference, this is the type of the `oninput` field for a `div`, which allows the `async` closure

![image](https://github.com/Satellite-im/Uplink/assets/8636602/c45549d7-1443-415d-bdbd-4533f194d237)